src/devices/RelayBoard/RelayBoard.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Iot.Device.RelayBoard
+{
+    /// <summary>
+    /// A board containing some <see cref="Relay"/> devices.
+    /// </summary>
+    /// <example>
+    /// Create a relay board, with a relay:
+    /// <code>
+    /// RelayBoard board = new RelayBoard();
+    /// board.CreateRelay(pin: 1);
+    /// </code>
+    /// </example>
+    public class RelayBoard : IDisposable, IEnumerable<Relay>, IReadOnlyCollection<Relay>, IReadOnlyDictionary<int, Relay>
+    {
+        private readonly GpioController _controller;
+        private readonly Dictionary<int, Relay> _relays = new ();
+        private readonly bool _shouldDispose;
+
+        /// <summary>
+        /// Gets the type of relay board.
+        /// </summary>
+        public RelayType Type { get; init; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RelayBoard"/> class.
+        /// </summary>
+        public RelayBoard(RelayType relayType = RelayType.NormallyClosed, GpioController? gpioController = null, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)
```

Comment:
Same comments from Relay.cs:
Is it possible to not default relay type? If this is going to become part of the framework, I suspect whatever is the default is always going to be the opposite of what is needed in someone's project